microclient.go

```diff
@@ -0,0 +1,11 @@
+// microclient provides opinionated helpers for REST clients.
+package microclient
+
+import (
+	"github.com/go-resty/resty"
+)
+
+// Do takes a resty request function and executes it, adding metrics.
+func Do(requestFunc func(string) (resty.Response, error)) (resty.Response, error) {
```

Comment:
Added a stub, this will be filled / changed / removed / obliterated in the next PR.